celestia.go

```diff
@@ -81,23 +84,46 @@ func (c *CelestiaDA) Commit(daBlobs []da.Blob) ([]da.Commitment, error) {
 	return daCommitments, nil
 }
 
+// Submit submits the Blobs to Data Availability layer.
 func (c *CelestiaDA) Submit(daBlobs []da.Blob) ([]da.ID, []da.Proof, error) {
 	var blobs []*blob.Blob
+	var commitments []da.Commitment
 	for _, daBlob := range daBlobs {
 		b, err := blob.NewBlobV0(c.namespace, daBlob)
 		if err != nil {
 			return nil, nil, err
 		}
 		blobs = append(blobs, b)
+
+		commitment, err := types.CreateCommitment(&b.Blob)
+		if err != nil {
+			return nil, nil, err
+		}
+		commitments = append(commitments, commitment)
 	}
 	height, err := c.client.Blob.Submit(c.ctx, blobs, blob.DefaultSubmitOptions())
 	if err != nil {
 		return nil, nil, err
 	}
-	fmt.Println("succesfully submitted blobs", "height", height)
-	return nil, nil, nil
+	log.Println("successfully submitted blobs", "height", height)
+	ids := make([]da.ID, len(daBlobs))
+	proofs := make([]da.Proof, len(daBlobs))
+	for i, commitment := range commitments {
+		ids[i] = makeID(height, commitment)
+		proof, err := c.client.Blob.GetProof(c.ctx, height, c.namespace, commitment)
+		if err != nil {
+			return nil, nil, err
+		}
+		// TODO(tzdybal): does always len(*proof) == 1?
+		proofs[i], err = (*proof)[0].MarshalJSON()
```

Comment:
What if proof is `nil` or is an empty array? This conversion from a `*blob.Proof` type to a `da.Proof` is quite confusing imo. Ideally, this conversion should be part of the blob API itself. For now, we should either add more comments explaining what's going on here or extract this into a helper function. Also, why is the length of the proof always 1?

Reply:
The story here is following:
* blob API returns `*blob.Proof`
* `blob.Proof` is `[]*nmt.Proof`
* blob API returns objects, which makes perfect sense - devs don't bother with (de)serialization (good DX)
* go-da API is generic - works on `[]byte`, so we need to serialize proof from blob API
* this project is an adapter - this is exactly the place where we should do this kind of "glue code"
* `*mnt.Proof` already implements `json.Marshaller` and `json.Unmarshaller` interfaces, so it was low hanging fruit

If proof is `nil` or empty, this should be signalized by `err != nil`.

Another issue is - what if `len(proof) > 1` - and this probably can happen with bigger blobs. I'll create a followup issues for this (as this should be covered by tests in go-da and fixed in implementation here).